File: tests/parsers/samples/code/php/complex.php
<?php

namespace Samples;

class ComplexMetrics
{
    public function __construct(private float $scale = 1.0)
    {
    }

    public function distance(int $a, int $b): float
    {
        return abs($a - $b) * $this->scale;
    }

    public function rootMeanSquare(array $values): float
    {
        if (count($values) === 0) {
            return 0.0;
        }
        $total = array_reduce($values, fn ($carry, $value) => $carry + $value * $value, 0.0);
        return sqrt($total / count($values));
    }

    public static function buildDefault(): self
    {
        return new self(0.88);
    }
}

File: tests/parsers/samples/code/lua/complex.lua
local complex = {}

function complex.distance(a, b)
    return math.abs(a - b)
end

function complex.root_mean_square(values)
    if #values == 0 then
        return 0
    end
    local total = 0
    for _, value in ipairs(values) do
        total = total + value * value
    end
    return math.sqrt(total / #values)
end

function complex.build_default()
    return { scale = 0.8 }
end

return complex
